zerver/tests/test_tornado.py

```diff
@@ -64,27 +64,26 @@ def get_new_ioloop(self) -> IOLoop:
     def get_app(self) -> Application:
         return create_tornado_application()
 
-    async def tornado_client_get(self, path: str, **kwargs: Any) -> HTTPResponse:
+    async def tornado_client_get(
+        self, path: str, skip_user_agent: bool, **kwargs: Any
+    ) -> HTTPResponse:
         self.add_session_cookie(kwargs)
-        kwargs["skip_user_agent"] = True
-        self.set_http_headers(kwargs)
+        self.set_http_headers(kwargs, skip_user_agent)
```

Comment:
Since `set_http_headers` does not expect `skip_user_agent` to be a key in `kwargs` and does not remove it from `skip_user_agent` anymore, it causes a runtime error when we call `self.http_client.fetch` later.
The only place where we call this function passes `skip_user_agent=True`, so I figured it to be clean to just reuse `skip_user_agent` as an argument here.

Reply:
That explains adding it as an argument to `set_http_headers`. But why add it as an argument to `tornado_client_get`?

Reply:
`tornado_client_get` has one caller; it does not set `skip_user_agent` (until you added it).

Reply:
I see. The caller set `skip_user_agent=True`, which was essentially noop when there was `kwargs["skip_user_agent"] = True`.

We can remove this argument (in the signature and for the caller) and have `self.set_http_headers(kwargs, True)` instead. Is this the intended behavior for this method?

Reply:
Ah, you’re right. That duplication happened in 27edc183306f8ac8d1ca61075980866e42ff3ab7 (#14020). Yes, that’s the intention; we need to unconditionally `skip_user_agent`, because `HTTP_USER_AGENT` is not valid syntax for the `tornado.httpclient` API.